libsolidity/formal/SMTEncoder.cpp

```diff
@@ -3124,12 +3124,7 @@ RationalNumberType const* SMTEncoder::isConstant(Expression const& _expr)
 	if (auto type = dynamic_cast<RationalNumberType const*>(_expr.annotation().type))
 		return type;
 
-	// _expr may not be constant evaluable.
-	// In that case we ignore any warnings emitted by the constant evaluator,
-	// as it will return nullptr in case of failure.
-	ErrorList l;
-	ErrorReporter e(l);
-	if (auto t = ConstantEvaluator::evaluate(e, _expr))
+	if (auto t = ConstantEvaluator::tryEvaluate(_expr))
```

Comment:
Now that we have the helper, we should use it in `StaticAnalyzer::visit(BinaryOperation)` as well. Though that needs a changelog entry and some tests so maybe it's better off done in a separate PR.

Reply:
Yes, I think this should be done in a separate PR.